library/core/src/cell.rs

```diff
@@ -1893,11 +1893,16 @@ impl<T: ?Sized + fmt::Display> fmt::Display for RefMut<'_, T> {
 /// uniqueness guarantee for mutable references is unaffected. There is *no* legal way to obtain
 /// aliasing `&mut`, not even with `UnsafeCell<T>`.
 ///
+/// `UnsafeCell` does nothing to avoid data races; they are still undefined behavior. If multiple
+/// threads have access to the same `UnsafeCell`, they must follow the usual rules of the
+/// [concurrent memory model]: conflicting non-synchronized accesses must be done via the `atomic` APIs.
```

Comment:
Is it possible to concretize "atomic APIs" to talk about particular symbols that the core library exports? That could alleviate some of the confusion I had in #129361, where it wasn't immediately clear to me what qualifies as an atomic operation. Maybe something like this?

```suggestion
/// [concurrent memory model]: conflicting non-synchronized accesses must be done via the [`core::sync::atomic`] APIs.
```

This is excluding compiler intrinsics as they are not intended to be used by stable external code.

Reply:
Sure, we can be fully explicit here.